protos/sources/vega/api/v1/core.proto

```diff
@@ -343,3 +348,84 @@ message LastBlockHeightResponse {
+  string party_id = 1;
+}
+
+// The statistics for a given spam policy
+message SpamStatistic {
+  // The current transaction count received from the party during this epoch for this policy
+  uint64 count_for_epoch = 1;
+  // The maximum number of transactions allowed for this policy in an epoch
+  uint64 max_for_epoch = 2;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string banned_until = 4;
+}
+
+// The voting statistics by proposal for a given party for the current epoch
+message VoteSpamStatistics {
+  // List of statistics for proposals voted on by the party
+  repeated VoteSpamStatistic statistics = 1;
+  // If blocked the timestamp when the party will be unblocked as RFC3339Nano
+  optional string banned_until = 2;
+}
+
+// The vote statistics for the voting spam policies
+// which are calculated as a ratio of the total votes
+// that have been rejected.
+message VoteSpamStatistic {
+  string proposal = 1;
+  // The current vote count received from the party for the given proposal during this epoch
+  uint64 count_for_epoch = 2;
+  // The maximum number of votes per proposal allowed in an epoch
+  uint64 max_for_epoch = 3;
+  // The effective minimum number of tokens required to vote on the proposal
+  string min_tokens_required = 4;
+}
+
+// The PoW state for a given block
+message PoWBlockState {
+  // Block height for the current Proof of Work state statistics
+  uint64 block_height = 1;
+  // Hash of the current block
+  string block_hash = 2;
+  // Total number of transactions seen in the block
+  uint64 transactions_seen = 3;
+  // The minimum expected difficulty for the next transaction in the block
+  // This is only provided if increasing difficulty is required
+  optional uint64 expected_difficulty = 4;
+  // The hashing function used to calculate the block hash
+  string hash_function = 5;
+}
+
+// Proof of work statistics for a party
+message PoWStatistic {
+  // Block state for each block in scope for PoW calculation
+  repeated PoWBlockState block_states = 1;
+  // PoW banned until timestamp as RFC3339Nano
+  optional string banned_until = 2;
+}
+
+// The complete spam statistics captured for a given party
+message SpamStatistics {
+  // Statistics for proposal transactions made by the party
+  SpamStatistic proposals = 1;
+  // Statistics for delegation transactions made by the party
+  SpamStatistic delegations = 2;
+  // Statistics for transfer transactions made by the party
+  SpamStatistic transfers = 3;
+  // Statistics for node announcement transactions made by the party
+  SpamStatistic node_announcements = 4;
+  // Statistics for proposal votes made by the party
+  VoteSpamStatistics votes = 5;
+  // Statistics for proof of work difficulty observed per block for the party
+  PoWStatistic pow = 6;
+}
+
+// A response containing all the spam statistics of a party for the current epoch
+message GetSpamStatisticsResponse {
+  SpamStatistics statistics = 1;
```

Comment:
If I understand this PR correctly, we would replace our current call to the last block information by this API. Then we need to return the ChainID in the payload.

Reply:
I don't think it is meant to be used in the way you describe - at least not explicitly.  With small changes it could, but as is, it only gives you statistics about blocks you already have transactions for, it won't tell you what the current block is nor what its hash is nor what the relevant pow params are.

Reply:
I don't think you can currently do that with the current implementation and I'm not aware that this was a requirement, those as commented above, it could be but would require some changes.

Reply:
Alright I get it. We need both calls then.

Reply:
not necessarily, I like the idea - I think all we need for that is to always add to this the last block. But we may do it separately if decided it's desired.